test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -532,6 +532,12 @@ double UcxContext::get_time() {
     return tv.tv_sec + (tv.tv_usec * 1e-6);
 }
 
+UcxConnection& UcxContext::get_connection(uint64_t id) const {
+    conn_map_t::const_iterator it = _conns.find(id);
+    assert(it != _conns.end());
+    return *it->second;
+}
```

Comment:
if stats map key was pointer, would not need to expose this function. for active messages, we don't really need to maintain connection id in ucx_wrapper.

Reply:
it's called 2 times per reporting, I would avoid another indexing

Reply:
mapping by pointer or by id should not really matter for performance
but is it possible that in case of AM, the id will not be initialized?

Reply:
in case of AM, id is an ep pointer, we should not have statistics for such connections (failure on start)

Reply:
connection id is internal to ucx_wrapper. IMO io_demo.cc should map by connection pointer. for performance, it doesn't really matter which hash key to use.

Reply:
I see your point, seems like we need to add another "Connection" class to expose public connection interface and keep UcxConnection for wrapper usage only

Reply:
for this PR, i think better make minimal changes: map by connection ptr to avoid exposing connection internal id